deps/ncrypto/ncrypto.cc

```diff
@@ -212,10 +212,25 @@ Buffer<void> DataPointer::release() {
 DataPointer DataPointer::resize(size_t len) {
   size_t actual_len = std::min(len_, len);
   auto buf = release();
-  if (actual_len == len_) return DataPointer(buf.data, actual_len);
-  buf.data = OPENSSL_realloc(buf.data, actual_len);
-  buf.len = actual_len;
-  return DataPointer(buf);
+  if (actual_len == len_) {
+    // Retain the secure_ heap flag
+    return DataPointer(buf.data, actual_len, secure_);
```

Comment:
IMO this `secure_` flag is problematic with resizes because secure memory comes from a different heap and should not be mixed with OPENSSL_realloc.

I believe it's a theoretical problem right now because secure buffers are never resized but it should either be a runtime error or a fatal error.

Reply:
Perhaps the simplest thing to do is to add this as the first line of resize:
```c++
if (secure_) return DataPointer();
```

Reply:
@bnoordhuis updated and pushed the branch based on this review. Kindly review the new commit.